src/zenml/enums.py

```diff
@@ -38,11 +38,15 @@ class StepRunInputArtifactType(StrEnum):
     MANUAL = "manual"  # manually loaded via `zenml.load_artifact()`
 
 
-class StepRunOutputArtifactType(StrEnum):
-    """All possible types of a step run output artifact."""
+class ArtifactSaveType(StrEnum):
+    """All possible method types of how artifact versions can be saved."""
 
-    DEFAULT = "default"  # output of the current step
+    STEP_OUTPUT = "default"  # output of the current step
     MANUAL = "manual"  # manually saved via `zenml.save_artifact()`
+    PREEXISTING = "preexisting"  # register via `zenml.register_artifact()`
+    EXTERNAL = (
```

Comment:
I'm now wondering about all these types: Are they really different from one another? What differentiates an `ExternalArtifact` from one saved with `save_artifact` for example? My current gut feeling is to even get rid of `ExternalArtifact` entirely and replace it by telling people to call `save_artifact` instead.

Also, why do we need to differentiate between `PREEXISTING` and `MANUAL` here, isn't this essentially the same thing as well?

Reply:
Yes, we can consider deprecating the `ExternalArtifact`, but this is not related to this ticket, I assume.

Regarding `PREEXISTING` and `MANUAL`, I would still keep it separate, since the `PREEXISTING` hacks how we store the data (paths ZenML creates are not respected anymore), which might be dangerous to manipulate during some housekeeping runs. Not the case as of now, but it will be safer to split those for future maintainability, IMO.